returnn/tf/layers/rec.py

```diff
@@ -5041,8 +5126,8 @@ def _create_search_beam(cls, name, beam_size, sources, network):
       name="%s%s" % (network.get_absolute_name_prefix(), name))
 
   @classmethod
-  def get_out_data_from_opts(cls, name, sources, target, network,
-                             beam_size, search=NotSpecified, scheduled_sampling=False, cheating=False, **kwargs):
+  def get_out_data_from_opts(cls, name, sources, target, network, beam_size, search=NotSpecified,
+                             scheduled_sampling=False, cheating=False, prefix_target=None, **kwargs):
```

Comment:
Instead of `prefix_target`, this could be a layer, like `prefix`. And you could directly refer to the whole sequence. So you would have `prefix="base:data:prefix"` or so in your config. This would avoid the whole padding logic.

Reply:
I don't understand. You mean `ChoiceLayer` gets the whole prefix and selects the label of the current timestep itself? Or would a layer somehow handle the padding for me?